src/utils/url/updateUrlQueryParam.ts

```diff
@@ -0,0 +1,28 @@
+import { parseQueryString, stringifyQuery } from "utils";
+
+export const updateUrlQueryParam = (
+  urlSearchParam: string,
+  inputValue: string | string[] | null,
+  search: string,
+  replace: (path: string) => void,
+  pathname: string,
+  sendAnalyticsEvent: (filterBy: string) => void,
+  resetPage?: boolean
+) => {
+  const urlParams = parseQueryString(search);
+
+  if (!inputValue) {
+    delete urlParams[urlSearchParam];
+  } else {
+    urlParams[urlSearchParam] = inputValue;
+  }
+
+  const nextQueryParams = stringifyQuery({
+    ...urlParams,
+    ...(resetPage && { page: 0 }),
+  });
+
+  replace(`${pathname}?${nextQueryParams}`);
+
+  sendAnalyticsEvent(urlSearchParam);
```

Comment:
I think this could be even more useful if we can somehow include the component id that the user interacted with to update the URL. I hope if we do this, then we don't have to call `sendAnalytics` in the input onChange handler.

Reply:
analytics will be done in a different PR. please only review what i listed in the summary

Reply:
also, it's not in the input onChange handler, it's in the update function, which is only called when the user clicks the "Search" button. This is exactly when and where the analytics event should be called.

Reply:
for `useFilterInputChangeHandler` , where this function is called in the input onChange handler,
1. the analytics event must be sent onChange because onChange is when the url is updated and filters are dispatched in a query
2. `useFilterInputChangeHandler` will soon be deprecated when all table filters are replaced with the hooks i introduce in this PR